### PR TITLE
MORE linalg tests!

### PR DESCRIPTION
**Pull request by kshyatt**
Removed a dead `convert`. Finally figured out how to test `A_ldiv_B!` for `AbstractVector`s. Finished testing `blas` and `svd`. Added a LAPACK test and finished the `schur` tests.

### CODE REVIEW

**Comment by kshyatt** [comment]
Tried this on latest master on `julia.mit.edu` and the LAPACK tests passed. Anyone have any thoughts?

File: test/linalg/diagonal.jl [hunk 1]
```diff
@@ -2,12 +2,45 @@
...
 import Base.LinAlg: BlasFloat, BlasComplex, SingularException
+import Base: size, linearindexing, LinearSlow, getindex, length, setindex!, ndims
 
 debug = false
 
 n=12 #Size of matrix problem to test
 srand(1)
 
+#we'll use this to test A_ldiv_B!
+type SparseVec{T} <: AbstractVector{T}
```
Comment (andreasnoack) [inline test/linalg/diagonal.jl]: Have you tried with `sub(ones(3), [1,2,3])`? It's not a `StridedVector`.

Comment (kshyatt) [inline test/linalg/diagonal.jl]: I couldn't get it to work and @vtjnash suggested this. I can redo the tests if necessary.

Comment (andreasnoack) [inline test/linalg/diagonal.jl]: I saw your discussion, but only this morning. It just seems a bit overkill to define a whole new type for this if an existing type actually covers this. Isn't it the method below you are interested in testing?
```jl
julia> @which A_ldiv_B!(Diagonal([1,2,3.0]), sub(randn(3), [1,2,3]))                       
A_ldiv_B!{T}(D::Base.LinAlg.Diagonal{T}, v::AbstractArray{T,1}) at linalg/diagonal.jl:99
```

Comment (kshyatt) [inline test/linalg/diagonal.jl]: Yes. I can change the test. Do we have any idea what's up with the LAPACK fail?

Comment (andreasnoack) [inline test/linalg/diagonal.jl]: Not really. I'll take a look.

**Pull request merged**